cli/core/pkg/distribution/artifact.go

```diff
@@ -80,6 +80,28 @@ func (aMap Artifacts) Fetch(version, os, arch string) ([]byte, error) {
 		"arch:%s", version, os, arch)
 }
 
+// FetchTest the test binary for a plugin version.
+func (aMap Artifacts) FetchTest(version, os, arch string) ([]byte, error) {
+	a, err := aMap.GetArtifact(version, os, arch)
+	if err != nil {
+		return nil, err
+	}
+
+	if a.Image != "" {
+		return artifact.NewOCIArtifact(a.Image).FetchTest()
+	}
+	if a.URI != "" {
+		u, err := artifact.NewURIArtifact(a.URI)
+		if err != nil {
+			return nil, err
+		}
+		return u.FetchTest()
+	}
+
+	return nil, errors.Errorf("invalid artifact for version:%s, os:%s, "+
```

Comment:
I'm just curious: Are you splitting the content of Errorf into two lines just for readability?

Reply:
Nope. Might be an oversight. I will fix this as part of addressing comments.